### PR TITLE
@damassi: Add bundle splitting via factor-bundle

### PR DESCRIPTION
**Pull request by craigspaeth**
This uses [factor-bundle](https://github.com/substack/factor-bundle) to do bundle splitting and moves the asset generation script from the [ezel-assets](github.com/artsy/ezel-assets) module into a scripts/assets.sh file.

This results in roughly the same total size asset packages but the benefit is it creates a large common.js package that can be better re-used/cached across pages. It also appears to compile assets faster. Building on top of this we could look into things like [tree shaking](https://github.com/nolanlawson/rollupify) and a nicer way to organize/load JS packages per-page so it doesn't rely on manually grouping them and loading by [route regexes](https://github.com/artsy/force/blob/master/desktop/assets/auctions.coffee#L14).

### CODE REVIEW

**Comment by damassi** [comment]
This is an in-general question: was testing this locally as simple as setting `NODE_ENV=production` and booting Force?

**Comment by craigspaeth** [comment]
Not quite—I had to slightly modify the script to output non-gzipped assets, [add a line](https://github.com/artsy/force/blob/master/lib/setup.js#L181) mounting the ./public folder as static middleware, and set `NODE_ENV`. It's possible we'd want to bake some of this into the codebase to make it easier to test locally.

File: scripts/assets.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  -t caching-coffeeify \
+  -t jadeify \
+  -t uglifyify \
+  -p [ factor-bundle -o 'uglifyjs | gzip > public/assets/`basename $FILE .coffee`.js.jgz' ] \
+  | uglifyjs | gzip > public/assets/common.js.jgz
+stylus \
+  $(find desktop/assets mobile/assets -name '*.styl') \
+  --compress \
+  -o public/assets
+gzip -S .cgz $(find public/assets -name '*.css')
```
Comment (damassi) [inline scripts/assets.sh]: It seems nicer to be able to control the asset pipeline directly from within Force

**Comment by damassi** [comment]
As a general housekeeping thing, would it be simple enough to add some kind of `TEST_PRODUCTION` flag (or whatever) into this PR so that I could give it a spin? When I was trying to fix the React `console.error` message, being unable to test locally was a showstopper.

**Comment by damassi** [comment]
Speaking of which, that error is still present, which makes me think that `envify` is indeed needed: https://www.artsy.net/auction/phillips-modern-masters. 

(Will follow up with Sarah about only rendering `redux-logger` actions in dev.)

**Comment by craigspaeth** [comment]
Actually @damassi that is likely because we didn't use `export NODE_ENV=` in the script. I'll update that [here](https://github.com/artsy/force/pull/1311/files#diff-917795c31c7a7e8f3ffe6745cf8a116eR5).

**Comment by craigspaeth** [comment]
Okay, updated to put `NODE_ENV=production browserify` in the asset script and updated the asset script and public folder glob to allow for testing production assets locally via `yarn assets && NODE_ENV= yarn start`.

**Comment by damassi** [comment]
Awesome, thanks @craigspaeth - being able to run an app locally in all of its various states makes things so much less anxious 👍

**Comment by damassi** [comment]
@craigspaeth, one last little thing: could you update https://github.com/artsy/force/blob/master/CONTRIBUTING.md#run-force with the new command above?

**Pull request merged**